src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

```diff
@@ -696,14 +695,73 @@ private async Task SubscribeToReturnResultsAsync()
             }
 
             var ros = completion.CompletionMessage;
-            var parseSuccess = protocol.TryParseMessage(ref ros, _clientResultsManager, out var hubMessage);
-            Debug.Assert(parseSuccess);
+            HubMessage? hubMessage = null;
+            bool retryForError = false;
+            try
+            {
+                var parseSuccess = protocol.TryParseMessage(ref ros, _clientResultsManager, out hubMessage);
+                retryForError = !parseSuccess;
+            }
+            catch
+            {
+                // Client returned wrong type? Or just an error from the HubProtocol, let's try with RawResult as the type and see if that works
+                retryForError = true;
```

Comment:
Why do we have to handle `!parseSuccess` and throwing the same way? Are our hub protocols not consistent on how they treat deserialization errors? Can we make the protocols swallow the `FormatExceptions` or whatever and always return false in these cases insetead?

Given that our protocols already need to be aware of `RawResult` for this feature to work at all, can we do one better and just make the protocols do this and call `CompletionMessage.WithError` themselves rather than mucking up the `RedisHubLifetimeManager`? It feels like a protocol concern when deserializing a client result fails because it doesn't match the Type provided by the IInvocationBinder.

Reply:
> Are our hub protocols not consistent on how they treat deserialization errors?

There are other protocol implementations, so we need to be resilient to their quirks.

> Given that our protocols already need to be aware of RawResult for this feature to work at all, can we do one better and just make the protocols do this and call CompletionMessage.WithError themselves rather than mucking up the RedisHubLifetimeManager?

This is something we can do in our protocol implementations, but I think all of this code would still need to stay for other implementations.